commit 05a318b4fc13f7eef85e22780a274cb206664533
Author: Jan Stancek <jstancek@redhat.com>
Date:   Tue Nov 12 09:21:33 2024 +0100

    tools: ynl: extend CFLAGS to keep options from environment
    
    Package build environments like Fedora rpmbuild introduced hardening
    options (e.g. -pie -Wl,-z,now) by passing a -spec option to CFLAGS
    and LDFLAGS.
    
    ynl Makefiles currently override CFLAGS but not LDFLAGS, which leads
    to a mismatch and build failure:
            CC sample devlink
      /usr/bin/ld: devlink.o: relocation R_X86_64_32 against symbol `ynl_devlink_family' can not be used when making a PIE object; recompile with -fPIE
      /usr/bin/ld: failed to set dynamic section sizes: bad value
      collect2: error: ld returned 1 exit status
    
    Extend CFLAGS to support hardening options set by build environment.
    
    Signed-off-by: Jan Stancek <jstancek@redhat.com>
    Acked-by: Jakub Kicinski <kuba@kernel.org>
    Reviewed-by: Donald Hunter <donald.hunter@gmail.com>
    Link: https://patch.msgid.link/265b2d5d3a6d4721a161219f081058ed47dc846a.1731399562.git.jstancek@redhat.com
    Signed-off-by: Jakub Kicinski <kuba@kernel.org>

diff --git a/tools/net/ynl/lib/Makefile b/tools/net/ynl/lib/Makefile
index 2887cc5de530..94c49cca3dca 100644
--- a/tools/net/ynl/lib/Makefile
+++ b/tools/net/ynl/lib/Makefile
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 
 CC=gcc
-CFLAGS=-std=gnu11 -O2 -W -Wall -Wextra -Wno-unused-parameter -Wshadow
+CFLAGS += -std=gnu11 -O2 -W -Wall -Wextra -Wno-unused-parameter -Wshadow
 ifeq ("$(DEBUG)","1")
   CFLAGS += -g -fsanitize=address -fsanitize=leak -static-libasan
 endif